### PR TITLE
Cult nerfs and hearing protection refactor repull

### PR DESCRIPTION
**Pull request by zeskorion**
## About The Pull Request
Nerfs a few cult items:

ritual dagger now has 12 throwforce, as opposed to 25(!!)
cult longsword can now be used by non-cultists, but this deals not insignificant stamina and brain damage per hit
lowers cult armor values across the board
mirror shield significantly nerfed: now no longer stuns, instead doing knockdown, has a lower block chance, can shatter from melee attacks, and now spawns with a fixed amount of non-recharging illusions
cult stun reflavored to occult whispers, can't stun through hearing protection

internal: totally refactored hearing protection, allowing more things to use hearing protection. expect a ling nerf using this in the future. Also fixes two items simply by existing; harm alarm and dance grenade (on socially awkward people) no longer bypass hearing protection

## Why It's Good For The Game

cult is a conversion antag where each cultist can very easily get traitor level gear, if not higher. this brings them a bit more in line (though, they are still exceptionally powerful,even still overpowered- but no longer downright unfair)

already passed vote, been sitting for near two months now. Fuckin merge it so i can use this hearing shit to make at least three changes
## Changelog
:cl:
balance: ritual dagger now has 12 throwforce, as opposed to 25
balance: cult longsword can now be used by non-cultists, but this deals not insignificant stamina and brain damage per hit
balance: lowers cult armor values across the board
balance: mirror shield significantly nerfed: now no longer stuns, instead doing knockdown, has a lower block chance, can shatter from melee attacks, and now spawns with a fixed amount of non-recharging illusions
balance: cult stun reflavored to occult whispers, can't stun through hearing protection
fix: harm alarm now cares about hearing protection
fix: dance grenade socially awkward now cares about hearing protection
tweak: headphones now provide protection from flashbang if worn on the ear slot
tweak: earmuffs now provide more robust hearing protection
code: hearing protection refactor
/:cl:

### CODE REVIEW

**Comment by Bastian0930** [comment]
can you just merge the hearing shit seperately? feel like it's out of scope of the PR

**Comment by zeskorion** [comment]
> can you just merge the hearing shit seperately? feel like it's out of scope of the PR

It's here because it's essential for the stun ti work

**Comment by qwertyquerty** [comment]
Vote is dead tied

**Comment by zeskorion** [comment]
> Vote is dead tied

it already won the vote before. this was a repull cuz the branch got old 
image

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 1]
```diff
@@ -24,6 +24,17 @@
 
 /mob/living/carbon/get_ear_protection()
 	var/number = ..()
+	if(istype(src.head, /obj/item/clothing/head))			//are they wearing something on their head
+		var/obj/item/clothing/head/HHP = src.head			//if yes gets the flash protection value from that item
+		number += HHP.bang_protect
+
+	if(istype(src.ears, /obj/item/radio/headset))		//headset
+		var/obj/item/clothing/glasses/RHP = src.ears
```
Comment (ike709) [inline code/modules/mob/living/carbon/carbon_defense.dm]: 🤔

Comment (zeskorion) [reply code/modules/mob/living/carbon/carbon_defense.dm]: pasting error, fixed

**Comment by ike709** [comment]
Why'd you replace the component with a var?

**Comment by zeskorion** [comment]
> Why'd you replace the component with a var?

the component quite literally only exists for the purpose of flashbangs. I replaced it for two reasons
1. so hearing protection is a bit more intuitive to code with
2. so hearing protection can by a dynamic number instead of something you either do or dont have

it's also already set up to be a number anyways, considering how it works with ears and such

**Comment by ike709** [comment]
Eh I like cult nerfs so I'll give it a pass.

**Pull request merged**